### PR TITLE
ZEN-2736 - wrap manipulations with validation markers in to a WorkspaceJob to batch changes.

### PR DESCRIPTION
**Pull request by tfesenko**
See https://github.com/ModelSolv/RepreZen/pull/1009 and
https://eclipse.org/articles/Article-Resource-deltas/resource-deltas.html
for details .

@andylowry @ghillairet can you please review?

### CODE REVIEW

File: com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java [hunk 1]
```diff
@@ -310,17 +313,23 @@ protected void validate(boolean onOpen) {
...
+                
+                @Override
+                public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+                    clearMarkers(file);
+                    validateYaml(file, (SwaggerDocument) document);
+                    validateSwagger(file, (SwaggerDocument) document, fileEditorInput);
+                    return Status.OK_STATUS;
+                }
+            }.schedule();
 		}
```
Comment (andylowry) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerEditor.java]: Indentation is off here

**Comment by andylowry** [comment]
@tfesenko Code review looks good to me, except that code indentation appears to be off.

**Comment by andylowry** [comment]
@tfesenko Thanks, pass code review

**Comment by ghillairet** [comment]
Looks good.

**Pull request merged**